### PR TITLE
[type:feat]shenyu-client add discovery config registration 

### PR DESCRIPTION
**Pull request by xcsnx**
Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-register-center/shenyu-register-common/src/main/java/org/apache/shenyu/register/common/config/ShenyuDiscoveryConfig.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+ * The type Discovery config.
+ */
+public class ShenyuDiscoveryConfig extends PropertiesConfig {
+    private String name;
+
+    private String type;
+
+    private String serverList;
+
+    private Properties props = new Properties();
```
Comment (ywj1352) [inline shenyu-register-center/shenyu-register-common/src/main/java/org/apache/shenyu/register/common/config/ShenyuDiscoveryConfig.java:32]: PropertiesConfig has Properties props  no need to add agin

Comment (xcsnx) [reply shenyu-register-center/shenyu-register-common/src/main/java/org/apache/shenyu/register/common/config/ShenyuDiscoveryConfig.java:32]: Thank you for your suggestion. I will improve it

**Comment by ywj1352** [comment]
looks good

File: shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/register/ClientDiscoveryConfigRefreshedEventListener.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    protected DiscoveryConfigRegisterDTO buildDiscoveryConfigRegisterDTO(final ShenyuDiscoveryConfig shenyuDiscoveryConfig) {
+        try {
+            return DiscoveryConfigRegisterDTO.builder()
+                    .name(shenyuDiscoveryConfig.getName())
+                    .serverList(shenyuDiscoveryConfig.getServerList())
+                    .props(shenyuDiscoveryConfig.getProps())
+                    .type(shenyuDiscoveryConfig.getType())
+                    .build();
+        } catch (ShenyuException e) {
+            throw new ShenyuException(e.getMessage() + "please config ${shenyu.discovery} in xml/yml !");
```
Comment (moremind) [inline shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/register/ClientDiscoveryConfigRefreshedEventListener.java:52]: remove e.getMessage, throw the custom message and add log there

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/register/ClientDiscoveryConfigRefreshedEventListener.java (hunk 2 above)
Comment (moremind) [reply shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/register/ClientDiscoveryConfigRefreshedEventListener.java:52]: why the builder throw exception?

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/register/ClientDiscoveryConfigRefreshedEventListener.java (hunk 2 above)
Comment (xcsnx) [reply shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/register/ClientDiscoveryConfigRefreshedEventListener.java:52]: Thank you. This is indeed unreasonable, I will modify it

**Review by ywj1352 — APPROVED** [review]

**Review by moremind — APPROVED** [review]

**Pull request merged**